tfx/orchestration/experimental/kubernetes/kubernetes_dag_runner.py

```diff
@@ -0,0 +1,378 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Definition of kubernetes TFX runner."""
+
+import absl
+import datetime
+import re
+import time
+from typing import Optional, Text, Type
+
+from ml_metadata.proto import metadata_store_pb2
+from tfx.dsl.component.experimental import container_component
+from tfx.components.base import base_node
+from tfx.orchestration import data_types
+from tfx.orchestration import metadata
+from tfx.orchestration import pipeline as tfx_pipeline
+from tfx.orchestration import tfx_runner
+from tfx.orchestration.config import base_component_config
+from tfx.orchestration.config import config_utils
+from tfx.orchestration.config import pipeline_config
+from tfx.orchestration.kubeflow import node_wrapper
+from tfx.orchestration.kubeflow import utils
+from tfx.orchestration.launcher import base_component_launcher
+from tfx.orchestration.launcher import in_process_component_launcher
+from tfx.orchestration.launcher import kubernetes_component_launcher
+from tfx.utils import json_utils, kube_utils
+from google.protobuf import json_format
+from kubernetes import client
+import json
+
+_CONTAINER_COMMAND = [
```

Comment:
How is this container environment created?  Can we check in the Dockerfile with instructions on how to use this?  It would be better if we can abstract this away (e.g. specify this as the entrypoint in the Dockerfile).

Reply:
The container needs to contain a dev version of TFX that has the newest features in KubernetesDagRunner. Otherwise, the defualt TFX image would be sufficient. I've checked in the Dockerfile which downloads the test version. Alternatively, we may ask users to modify the official docker image in tfx/tools/docker to install this test version. Ideally when we release the feature, the container image would be updated to contain the kubernetes orchestrator.